framework/src/base/FEProblemBase.C

```diff
@@ -2859,6 +2861,25 @@ FEProblemBase::executeControls(const ExecFlagType & exec_type)
 }
 
 void
+FEProblemBase::executeSamplers(const ExecFlagType & exec_type)
+{
+  // TODO: This should be done in a threaded loop, but this should be super quick so for now
+  // do a serial loop.
+  Moose::perf_log.push("executeSamplers()", "Execution");
```

Comment:
I've putting putting this perf_log push/pop pairs inside the `(!objects.empty()` check so that they only show up when active. It cuts down on several rows of zeros.